src/controllers/notifications.js

```diff
@@ -71,6 +73,16 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
+
+  const verifySignature = this.config.get('notifications.must_verify')
+  if (verifySignature) {
+    const result = model.verifySignature(notification, this.config)
+    if (!result.valid) {
+      log.warn('Signature verification failed: ' + result.error)
+      throw new UnprocessableEntityError('Notification failed signature verification')
```

Comment:
> There's no way to know anything is failing unless there's a separate process parsing the logs.

This is also true in our current implementation of the ledger. Ledger gets error, logs, retries notification, gets error, logs, retries, etc. As far as I know, we don't have a mechanism for informing RC of a failed notification between the ledger and the connector. The errors bounce back and forth between the two servers and are confined to the logs.

The most obvious way this issue could arise is if the connector/ledger are configured with different keys. This is an exceptional case, and should probably result in an exception, but we should handle this, rather than blindly retrying the notification, as we currently do.

My understanding is that notification retries address the situation in which one server is down temporarily or, for whatever reason, the notification didn't reach the recipient, which is not the case here. In this case, it seems unlikely that retrying the notification will result in a different outcome, since notifications are signed the same way each time.

One way to address this would be to use a more specific error than `UnprocessableEntityError` (maybe `InvalidSignatureError` or something), so that we can treat instances of this error differently. I think the above code works for the PR, but we shouldn't lose sight of this discussion.